pkgs/by-name/ck/ckb/package.nix

```
@@ -0,0 +1,63 @@
+{
+  cmake,
+  fetchFromGitHub,
+  lib,
+  ckb,
+  openssl,
+  pkg-config,
+  rustPlatform,
+  stdenv,
+  testers,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ckb";
+  version = "0.202.0";
+
+  src = fetchFromGitHub {
+    owner = "nervosnetwork";
+    repo = "ckb";
+    tag = "v${version}";
+    hash = "sha256-LE+w5RHkDk+1Zu2RZgPyHbHJ92C1oAxGEcVc3Qo2lQc=";
+  };
+
+  cargoHash = "sha256-R7XV//e6n6afbad7HgwlcNwQRdO0xFZ0UGy1SU294Fs=";
+
+  buildFeatures = [
+  ]
+  ++ lib.optionals (stdenv.hostPlatform.isDarwin) [ "portable" ];
+
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+  ];
+
+  OPENSSL_NO_VENDOR = true;
+
+  __darwinAllowLocalNetworking = true;
+
+  passthru = {
+    tests.version = testers.testVersion {
+      package = ckb;
+      command = "ckb --version";
+      version = "ckb ${ckb.version}";
+    };
+  };
```

```suggestion
  doInstallCheck = true; 
  nativeInstallCheckInputs = [ versionCheckHook ];
```
A bit more convenient, because it's involved in the build process